widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionActivityWatcher.kt

```diff
@@ -28,16 +29,13 @@ internal class EngagementCompletionActivityWatcher(
 
         when {
             event.consumed -> Logger.d(TAG, "skipping.. event: ${event.value} is already consumed")
-
-            /*
-            * This state finishes all SDK activities even if the application is in the background(activity == null),
-            * to be sure that everything related to the queueing or engagement is released.
-            * */
-            state is EngagementCompletionState.QueuingOrEngagementEnded -> event.consume { finishActivities() }
-            activity == null || activity.isFinishing -> Logger.d(TAG, "skipping.. activity is null or finishing")
+            activity == null || activity.isFinishing -> {
+                Logger.d(TAG, "skipping.. activity is null or finishing")
+                if (state is EngagementCompletionState.EngagementEnded) event.consume { finishActivities() }
```

Comment:
Let me also double check here because I don't know the reason for removing the comment and making changes here. Before the change we were checking for QueuingOrEngagementEnded. After the change we check for Engagement only (not interested in queueing anymore). Is this intended?

Reply:
> I don't know the reason for removing the comment
I have re-read it multiple times only to understand that it is not really useful here. It literally explained what is obvious from one line code with some additional details that are not really helpful

Reply:
> Before the change we were checking for QueuingOrEngagementEnded. After the change we check for Engagement only (not interested in queueing anymore). Is this intended?

This is a bit harder to explain as `QueuingOrEngagementEnded` is removed from the list and variation of `event.consume { finishActivities() }` is called in many different cases. I have tested this logic quite a lot and have not detected any problems with new solution